src/main/kotlin/no/nav/familie/ef/sak/opplysninger/personopplysninger/EndringerIPersonOpplysningerService.kt

```diff
@@ -0,0 +1,68 @@
+package no.nav.familie.ef.sak.opplysninger.personopplysninger
+
+import no.nav.familie.ef.sak.behandling.BehandlingService
+import no.nav.familie.ef.sak.behandling.Saksbehandling
+import no.nav.familie.ef.sak.behandling.domain.BehandlingStatus
+import no.nav.familie.ef.sak.felles.util.harGåttAntallTimer
+import no.nav.familie.ef.sak.infrastruktur.exception.feilHvis
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.domene.Grunnlagsdata
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.domene.GrunnlagsdataMedMetadata
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.dto.Endringer
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.dto.EndringerIPersonopplysningerDto
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.time.LocalDateTime
+import java.util.UUID
+
+@Service
+class EndringerIPersonOpplysningerService(
+    private val behandlingService: BehandlingService,
+    private val grunnlagsdataService: GrunnlagsdataService,
+    private val personopplysningerService: PersonopplysningerService
+) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    fun hentEndringerPersonopplysninger(behandlingId: UUID): EndringerIPersonopplysningerDto {
+        val behandling = behandlingService.hentSaksbehandling(behandlingId)
+        feilHvis(behandling.status == BehandlingStatus.FERDIGSTILT) {
+            "Kan ikke hente endringer for behandling med status=${behandling.status}"
+        }
+        val grunnlagsdata = grunnlagsdataService.hentLagretGrunnlagsdata(behandlingId)
+        return hentEndringerPersonopplysninger(behandling, grunnlagsdata)
+    }
+
+    private fun hentEndringerPersonopplysninger(
+        behandling: Saksbehandling,
+        grunnlagsdata: Grunnlagsdata,
+    ): EndringerIPersonopplysningerDto {
+        val skalSjekkeDataFraRegisteret = grunnlagsdata.oppdaterteDataHentetTid.harGåttAntallTimer(4)
```

Comment:
oppdaterteDataHentetTid <-- Kan denne bli litt pussig for alt som skjer etter deploy? Da finnes den jo ikke og er alltod "now" - følgelig skal man aldri sjekke mot registeret? 🤔

Reply:
Den settes til opprettet tid for grunnlagsdata på de som mangler den fra før